gensim/matutils.py

```diff
@@ -382,6 +383,112 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    Vec is considered to be in bag of words format if it is 2-tuple format.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    try:
+        id_, val_ = vec[0] # checking first value to see if it is in bag of words format by unpacking
+        id_, val_ = int(id_), float(val_)
+    except IndexError:
+        return True # this is to handle the empty input case
+    except Exception:
+        return False
+    return True
+
+
+def kullback_leibler(vec1, vec2, num_features=None):
+    """
+    A similarity metric between two probability distributions.
+    Returns a similarity in range <0,1> where values closer to 0 mean a higher similarity.
+    Uses the scipy.stats.entropy method to identify kullback_leibler convergence value.
+    If the distribution draws from a certain number of docs, that value must be passed.
+    """
+    if scipy.sparse.issparse(vec1):
+        vec1 = numpy.asarray(vec1.todense())
+    if scipy.sparse.issparse(vec2):
+        vec2 = numpy.asarray(vec2.todense()) # converted both the vectors to dense in case they were in sparse matrix 
+    if isbow(vec1) and isbow(vec2): # if they are in bag of words format we make it dense
+        if num_features != None: # if not None, make as large as the documents drawing from
+            max_len = max(len(vec1), len(vec2), num_features)
```

Comment:
Why this `max`? If the number of features is given, it should just be used.

Also, if `num_features` really is larger than `len(vec1)` and `len(vec2)`, it will have no effect on the result anyway. Adding extra zeros to the end of both vectors shouldn't change the outcome of `entropy`.